src/imagery/i.eodag/i.eodag.py

```diff
@@ -799,6 +829,47 @@ def products_compare(first, second):
     return search_result
 
 
+def skip_existing(output, search_result):
+    """Remove products that is already downloaded and saved in 'output' directory.
+
+    :param output: Output directory whose files will be compared with the scenes.
+    :type output: class'eodag.api.search_result.SearchResult'
+
+    :param search_results: EO products to be checked for existence in 'output' directory.
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    filtered_result = []
+    # Check for previously downloaded scenes
+    existing_files = [os.path.join(output, f) for f in os.listdir(output)]
+    if len(existing_files) <= 1:
+        return search_result
+    # TODO: Implement a way to check if a file is completely downloaded
+    #       or if it was interrupted
+    for scene in search_result:
+        existing_file = [
+            sfile for sfile in existing_files if scene.properties["title"] in sfile
+        ]
+        if existing_file:
+            creation_time = datetime.utcfromtimestamp(
+                os.path.getctime(existing_file[0])
+            )
+            ingestion_time = datetime.fromisoformat(
+                scene.properties["startTimeFromAscendingNode"]
+            ).replace(tzinfo=None)
+            if ingestion_time <= creation_time:
+                gs.message(
+                    _("Skipping scene: {} which is already downloaded.").format(
+                        scene.properties["title"]
+                    )
+                )
+                continue
+        filtered_result.append(scene)
+    return SearchResult(filtered_result)
```

Comment:
Could you do:
```suggestion
                search_result.remove(scene)
    return search_result
```
? That would avoid creating a new list, and I think _SearchResult_ is a _list_ object, so it should have a _.remove()_ method....

Reply:
Cool, that works yes.